### ISSUE TITLE
There are some doubts when training fedavg~

### ISSUE THREAD

**Issue by countingeyes** [TsingZ0/PFL-Non-IID#58]
Hi TsingZ0, thanks for your nice work!
I met some problems when training fedavg.
When I trained fedavg with 100 clients setting shards as 2,  the loss converges slowly，jitters violently and even drops by half at 400 epoch.  When I trained fedavg with 100 clients with iid setting, the loss and acc seem normal. I'm not sure if these are normal. What do you think of these？

![image](https://user-images.githubusercontent.com/75488849/222048737-d8701e86-8e92-4c96-b250-b256e11e4479.png)

The hyperparameters I used are as follows: 
--num_classes 10  --batch_size 64  --local_learning_rate 0.1 --lr_decay 0.998  --local_steps 2 --weight decay parameter 5e-4  --momentum 0.9  --join_ratio 0.1  --num_clients 100

**Comment by countingeyes** [comment]
The same thing occurs in fedbabu training progress. And fedrep meets the same.
![image](https://user-images.githubusercontent.com/75488849/222065771-a2d054de-dc04-42dd-8f70-6158741fcc5a.png)

**Comment by TsingZ0** [comment]
When setting `-lr 0.005 -lbs 10 -ls 1` and using 4-layer CNN, you can get the following figure (with format METHOD-DATASET-SETTING-NUM_CLIENTS-JOIN_RATIO)
![image](https://user-images.githubusercontent.com/24207672/224552283-f085f404-0cc5-49e9-beb6-7e7a73601924.png)

**Comment by countingeyes** [comment]
> When setting `-lr 0.005 -lbs 10 -ls 1` and using 4-layer CNN, you can get the following figure (with format METHOD-DATASET-SETTING-NUM_CLIENTS-JOIN_RATIO)

 I have tried your setting and it works well.   I’m truly grateful for your help！
